### PR TITLE
CLOUDP-248342: Add Amazon Linux 2023 support

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

This PR adds support for Amazon Linux 2023.

[CLOUDP-248342](https://jira.mongodb.org/browse/CLOUDP-248342)
## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: build/ci/repo_config.yaml [hunk 1]
```diff
@@ -241,6 +241,13 @@ repos:
     repos:
       - yum/amazon/2/mongodb-enterprise
 
+  - name: amazon2
```
Comment (fmenezes) [inline build/ci/repo_config.yaml:244]:
```suggestion
  - name: amazon2023
```

**Review by fmenezes — DISMISSED** [review]

File: build/package/docker/repo/amazonlinux2023.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM amazonlinux:2023
+
+ARG package
+ARG entrypoint
+ARG server_version
+ARG mongo_package
+ARG mongo_repo
+
+RUN printf "[${mongo_package}-${server_version}]\nname=MongoDB Repository\nbaseurl=${mongo_repo}/yum/amazon/2/${mongo_package}/${server_version}/\$basearch/\ngpgcheck=1\nenabled=1\ngpgkey=https://www.mongodb.org/static/pgp/server-${server_version}.asc\n" > /etc/yum.repos.d/${mongo_package}-${server_version}.repo
```
Comment (tibulca) [inline build/package/docker/repo/amazonlinux2023.Dockerfile:9]: should this be?
```suggestion
RUN printf "[${mongo_package}-${server_version}]\nname=MongoDB Repository\nbaseurl=${mongo_repo}/yum/amazon/2023/${mongo_package}/${server_version}/\$basearch/\ngpgcheck=1\nenabled=1\ngpgkey=https://www.mongodb.org/static/pgp/server-${server_version}.asc\n" > /etc/yum.repos.d/${mongo_package}-${server_version}.repo
```

Comment (fmenezes) [reply build/package/docker/repo/amazonlinux2023.Dockerfile:9]: good catch

Comment (blva) [reply build/package/docker/repo/amazonlinux2023.Dockerfile]: yeah, updated

**Review by tibulca — APPROVED** [review]

**Pull request merged**